src/test/java/edu/harvard/iq/dataverse/api/DatasetsIT.java

```diff
@@ -2583,5 +2583,48 @@ public void testFilesUnchangedAfterDatasetMetadataUpdate() throws IOException {
                 .body("data.latestVersion.files[0].directoryLabel", equalTo("code"));
         
     }
+
```

Comment:
Thanks for adding a test! The comment I'm leaving below is about FilesIT rather than DatasetsIT but can you please check if FilesIT.test_007_ReplaceFileUnpublishedAndBadIds is passing on your branch? Jenkins is reporting the following failure at https://jenkins.dataverse.org/blue/organizations/jenkins/IQSS-Dataverse-Develop-PR/detail/PR-8532/4/tests


java.lang.IllegalArgumentException: Cannot get property 'files' on null object
	at edu.harvard.iq.dataverse.api.FilesIT.test_007_ReplaceFileUnpublishedAndBadIds(FilesIT.java:755)
Caused by: java.lang.NullPointerException: Cannot get property 'files' on null object
	at edu.harvard.iq.dataverse.api.FilesIT.test_007_ReplaceFileUnpublishedAndBadIds(FilesIT.java:755)

Reply:
Yes, FilesIT.test_007_ReplaceFileUnpublishedAndBadIds also failed on my machine. The reason is optionalFileParams can be null and hence not to have tabIngest parameter. So I added check if  optionalFileParams is null then tabIngest is true.
Now the test is working.

But I see that Jenkins failed again. I do not have access to see what the error could be.

Reply:
testSemanticMetadataAPIs was fixed by PR #8557.